src/logic/map_objects/tribes/productionsite.cc

```diff
@@ -605,37 +599,84 @@ bool ProductionSite::warp_worker(EditorGameBase& egbase, const WorkerDescr& wdes
 	return false;
 }
 
+/**
+ * @return index in descr().working_positions() (slot requirement) and working_positions_ (current
+ * slot)
+ */
+std::pair<int32_t, int32_t> ProductionSite::find_worker(OPtr<Worker> w) {
+	auto wp = working_positions_->begin();
+	const BillOfMaterials& bom = descr().working_positions();
+	int32_t wp_index = 0;
+	int32_t bom_index = 0;
+	for (auto temp_wp = bom.begin(); temp_wp != bom.end(); ++temp_wp, ++bom_index) {
+		for (uint32_t j = temp_wp->second; j; --j, ++wp, ++wp_index) {
+			if (wp->worker == w) {
+				return std::pair<size_t, size_t>(bom_index, wp_index);
+			}
+		}
+	}
+	return std::pair<int32_t, int32_t>(-1, -1);
+}
+
+void ProductionSite::try_replace_worker(const Game* game,
+                                        DescriptionIndex worker_index,
+                                        WorkingPosition* wp) {
```

Comment:
Also needs a brief comment what the function does and what the parameters `worker_index` and `wp` mean